### PR TITLE
move amp script file as the last child of `head` and add required style tag and async attribute

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

File: examples/instagram.amp.html [hunk 1]
```diff
@@ -7,13 +7,13 @@
   <meta name="viewport" content="width=device-width,initial-scale=1,minimum-scale=1,maximum-scale=1,user-scalable=no,minimal-ui">
   <link href='https://fonts.googleapis.com/css?family=Questrial' rel='stylesheet' type='text/css'>
   <script custom-element="amp-instagram" src="https://cdn.ampproject.org/v0/amp-instagram-0.1.js" async></script>
-  <script src="https://cdn.ampproject.org/v0.js" async></script>
   <style>body {opacity: 0}</style><noscript><style>body {opacity: 1}</style></noscript>
```
Comment (dvoytenko) [inline examples/instagram.amp.html]: Move <style> right before <script>? It seems that you were doing in other examples...

**Comment by dvoytenko** [comment]
LGTM with comments.

**Comment by paulirish** [comment]
Small nit from me, though I'm just observing. I know some prefer to see the `async` attribute before the `src` attribute so its less of a surprise when reading the source.

**Comment by WebReflection** [comment]
> I know some prefer to see the async attribute before the src attribute so its less of a surprise when reading the source.

I totally agree with @paulirish for the nit ... yet somebody has to explain me why at Google you chaps write the underscore at **the end** of the property name and not upfront.

The argument that `VBScript` via old `jscript.dll` didn't accept it as prefix is invalid :P

**Comment by erwinmombay** [comment]
@paulirish @WebReflection applied recommendation to all our documentation and code, thanks!

**Comment by cramforce** [comment]
<3 LGTM

File: examples/instagram.amp.html (hunk 1 above)
Comment (erwinmombay) [inline examples/instagram.amp.html]: done.

**Comment by dvoytenko** [comment]
@WebReflection We use the suffixing underscore because we currently following JS guidelines for Closure: https://google.github.io/styleguide/javascriptguide.xml#Visibility__private_and_protected_fields_

We'll make a note of this in our dev docs. The reason we support Closure is mainly because we want to ensure that we could swap different compilers freely. We expect Closure to give us a solid type checking and also better obfuscation and dead-code elimination.

**Pull request merged**